### PR TITLE
Allow opt-in for Go race detector per SDK

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
This is a stop-gap until a more general solution is found. Some tests will cause an OOM when the race detector is enabled. For now, allow SDKs to opt in to the race detector if their tests don't cause an OOM.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/steps/build-test.yml [hunk 1]
```diff
@@ -70,7 +73,7 @@ steps:
     inputs:
       targetType: 'filePath'
       filePath: ./eng/scripts/run_tests.ps1
-      arguments: '${{ parameters.ServiceDirectory }} ${{ parameters.TestRunTime }}'
+      arguments: '${{ parameters.ServiceDirectory }} ${{ parameters.TestRunTime }} ${{ parameters.EnableRaceDetector }}'
```
Comment (jhendrixMSFT) [inline eng/pipelines/templates/steps/build-test.yml:76]: @benbp is there some magic I can do here to pass a proper PS `$true`? It's passing `True` which the PS script doesn't like.

Comment (jhendrixMSFT) [reply eng/pipelines/templates/steps/build-test.yml]: I tried adding a leading `$`, let's see if that works.

Comment (jhendrixMSFT) [reply eng/pipelines/templates/steps/build-test.yml]: It worked.

**Comment by jhendrixMSFT** [comment]
See https://github.com/Azure/azure-sdk-for-go/pull/20467 for history.

**Pull request merged**